logger.go

```diff
@@ -79,8 +90,13 @@ func (l *MatchingLogger) Write(p []byte) (int, error) {
 	}
 
 	// filter out already-timestamped logging from stdout
-	dateRegex := regexp.MustCompile(`^[0-9]{4}-[0-9]{2}-[0-9]{2}T`)
 	if dateRegex.Match(p) {
+		// TODO(RSDK-7895): these lines are sometimes multi-line.
+		if l.upload {
+			if parsed := parseLog(p); parsed.valid() && globalNetAppender != nil {
+				globalNetAppender.Write(parsed.entry(), nil) //nolint:errcheck,gosec
```

Comment:
was concerned that if I logged the error, it would end up in the same function and hang forever on the lock at L87 above

options:
- create a non-netappender logger for this kind of error? it wouldn't be logged to network, but it would end up in journalctl
- delayed call in goroutine with some kind of recursion limit so it won't get into 'an error logging an error' situations

Reply:
done in 5c81cbf6